Fails to build with Clang (due to -Werror)

Regressed by #4939. From [error log](https://github.com/yuzu-emu/yuzu/files/5619396/yuzu-qt5-s20201130.log):
```c++
src/input_common/mouse/mouse_input.cpp:60:16: error: implicit conversion changes signedness: 'int' to 'std::__1::array::size_type' (aka 'unsigned long') [-Werror,-Wsign-conversion]
    mouse_info[button_].mouse_origin = Common::MakeVec(x, y);
    ~~~~~~~~~~ ^~~~~~~
src/input_common/mouse/mouse_input.cpp:61:16: error: implicit conversion changes signedness: 'int' to 'std::__1::array::size_type' (aka 'unsigned long') [-Werror,-Wsign-conversion]
    mouse_info[button_].last_mouse_position = Common::MakeVec(x, y);
    ~~~~~~~~~~ ^~~~~~~
src/input_common/mouse/mouse_input.cpp:62:16: error: implicit conversion changes signedness: 'int' to 'std::__1::array::size_type' (aka 'unsigned long') [-Werror,-Wsign-conversion]
    mouse_info[button_].data.pressed = true;
    ~~~~~~~~~~ ^~~~~~~
src/input_common/mouse/mouse_input.cpp:91:16: error: implicit conversion changes signedness: 'int' to 'std::__1::array::size_type' (aka 'unsigned long') [-Werror,-Wsign-conversion]
    mouse_info[button_].tilt_speed = 0;
    ~~~~~~~~~~ ^~~~~~~
src/input_common/mouse/mouse_input.cpp:92:16: error: implicit conversion changes signedness: 'int' to 'std::__1::array::size_type' (aka 'unsigned long') [-Werror,-Wsign-conversion]
    mouse_info[button_].data.pressed = false;
    ~~~~~~~~~~ ^~~~~~~
src/input_common/mouse/mouse_input.cpp:93:16: error: implicit conversion changes signedness: 'int' to 'std::__1::array::size_type' (aka 'unsigned long') [-Werror,-Wsign-conversion]
    mouse_info[button_].data.axis = {0, 0};
    ~~~~~~~~~~ ^~~~~~~
6 errors generated.
```

--- a/src/input_common/mouse/mouse_input.cpp
+++ b/src/input_common/mouse/mouse_input.cpp
@@ -54,12 +54,13 @@ void Mouse::PressButton(int x, int y, int button_) {
     }
 
     int button = 1 << button_;
+    const auto button_index = static_cast<std::size_t>(button_);
     buttons |= static_cast<u16>(button);
     last_button = static_cast<MouseButton>(button_);
 
-    mouse_info[button_].mouse_origin = Common::MakeVec(x, y);
-    mouse_info[button_].last_mouse_position = Common::MakeVec(x, y);
-    mouse_info[button_].data.pressed = true;
+    mouse_info[button_index].mouse_origin = Common::MakeVec(x, y);
+    mouse_info[button_index].last_mouse_position = Common::MakeVec(x, y);
+    mouse_info[button_index].data.pressed = true;
 }
 
 void Mouse::MouseMove(int x, int y) {
@@ -86,11 +87,12 @@ void Mouse::ReleaseButton(int button_) {
     }
 
     int button = 1 << button_;
+    const auto button_index = static_cast<std::size_t>(button_);
     buttons &= static_cast<u16>(0xFF - button);
 
-    mouse_info[button_].tilt_speed = 0;
-    mouse_info[button_].data.pressed = false;
-    mouse_info[button_].data.axis = {0, 0};
+    mouse_info[button_index].tilt_speed = 0;
+    mouse_info[button_index].data.pressed = false;
+    mouse_info[button_index].data.axis = {0, 0};
 }
 
 void Mouse::BeginConfiguration() {

